File: binance4j-fiat/src/main/java/com/binance4j/fiat/dto/TransactionHistory.java
package com.binance4j.fiat.dto;

import java.util.List;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;

/**
 * Transaction history.
 * 
 * @param code    Code.
 * @param message Message.
 * @param data    Transactions.
 * @param total   Total.
 * @param success Success.
 */
@ApiModel("Transaction history.")
public record TransactionHistory(@ApiModelProperty("Code.") String code, @ApiModelProperty("Message.") String message,
		@ApiModelProperty("Transactions.") List<Transaction> data, @ApiModelProperty("Total.") int total,
		@ApiModelProperty("Success.") boolean success) {
}

File: connectors/src/main/kotlin/com/binance4j/fiat/dto/TransactionHistory.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.fiat.dto

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty

/**
 * Transaction history.
 *
 * @property code Code.
 * @property message Message.
 * @property data Transactions.
 * @property total Total.
 * @property success Success.
 */
@ApiModel("Transaction history.")
data class TransactionHistory(
	@ApiModelProperty("Code.")
	val code: String,
	@ApiModelProperty("Message.")
	val message: String,
	@ApiModelProperty("Transactions.")
	val data: List<Transaction>,
	@ApiModelProperty("Total.")
	val total: Int,
	@ApiModelProperty("Success.")
	val success: Boolean
)